py-polars/polars/lazyframe/frame.py

```diff
@@ -518,45 +529,88 @@ def _scan_python_function(
         return self
 
     @classmethod
+    @deprecate_function(
+        "Convert the JSON string to `StringIO` and then use `LazyFrame.deserialize`.",
+        version="0.18.12",
+    )
     def from_json(cls, json: str) -> Self:
         """
         Read a logical plan from a JSON string to construct a LazyFrame.
 
+        .. deprecated:: 0.18.12
+            This method is deprecated. Convert the JSON string to ``StringIO``
+            and then use ``LazyFrame.deserialize``.
+
         Parameters
         ----------
         json
             String in JSON format.
 
         See Also
         --------
-        read_json
+        deserialize
 
         """
-        bytes = StringIO(json).getvalue().encode()
-        file = BytesIO(bytes)
-        return cls._from_pyldf(PyLazyFrame.read_json(file))
+        return cls.deserialize(StringIO(json))
 
     @classmethod
-    def read_json(cls, file: str | Path | IOBase) -> Self:
+    @deprecate_renamed_function("deserialize", version="0.18.12")
+    @deprecate_renamed_parameter("file", "source", version="0.18.12")
+    def read_json(cls, source: str | Path | IOBase) -> Self:
```

Comment:
I found out that the `deprecate_renamed_methods` class decorator does not work on class methods, so we need to deprecate the old-fashioned way.